main.nf

```diff
@@ -99,35 +99,45 @@ process GATK_JOINTGENOTYPING {
 
 workflow {
 
-    // Create input channel from samplesheet in CSV format (via CLI parameter)
-    reads_ch = Channel.fromPath(params.reads_bam)
-                        .splitCsv(header: true)
-                        .map{row -> [row.id, file(row.reads_bam)]}
+    // Create input channel from BAM files
+    // We convert it to a tuple with the file name and the file path
+    // See https://www.nextflow.io/docs/latest/script.html#getting-file-attributes
+    bam_ch = Channel.fromPath(params.bams, checkIfExists: true)
+                        .map{bam -> [bam.simpleName, bam]}
+
+    
+    // Create reference channels using the fromPath channel factory
+    // The collect converts from a queue channel to a value channel
+    // See https://www.nextflow.io/docs/latest/channel.html#channel-types for details
+    ref_ch               = Channel.fromPath(params.reference, checkIfExists: true).collect()
+    ref_index_ch         = Channel.fromPath(params.reference_index, checkIfExists: true).collect()
+    ref_dict_ch          = Channel.fromPath(params.reference_dict, checkIfExists: true).collect()
+    calling_intervals_ch = Channel.fromPath(params.calling_intervals, checkIfExists: true).collect()
 
     // Create index file for input BAM file
-    SAMTOOLS_INDEX(reads_ch)
+    SAMTOOLS_INDEX(bam_ch)
 
     // Call variants from the indexed BAM file
     GATK_HAPLOTYPECALLER(
         SAMTOOLS_INDEX.out,
-        params.genome_reference,
-        params.genome_reference_index,
-        params.genome_reference_dict,
-        params.calling_intervals
+        ref_ch,
+        ref_index_ch,
+        ref_dict_ch,
+        calling_intervals_ch
     )
 
-    // Create a sample map of the output GVCFs
-    sample_map = GATK_HAPLOTYPECALLER.out.collectFile(){ id, gvcf, idx ->
-            ["${params.cohort_name}_map.tsv", "${id}\t${gvcf}\t${idx}\n"]
-    }
+    vcfs_in = GATK_HAPLOTYPECALLER.out
+        .map { id, vcf, idx ->
+            [ params.cohort_name, vcf, idx ]
+        }
+        .groupTuple()
```

Comment:
Hah I didn't realize you could do that. Yes this simplifies things enormously. But doesn't it make some other parts of the tutorial obsolete? Ie the bits that are essentially about propagating metadata (sample name etc since we only really needed it for the sample map stuff). 

Which is not a problem-- it may be that the right thing to do is to move that stuff out into a separate training module that's specifically about metadata. Maybe that connects to the sample sheet point--maybe for this training module we're like, ok let's assume we're just grabbing bams from a directory, and in a later module we can say ok now let's give you more fine grained control by using samplesheets, the metamap and all that good stuff.